### PR TITLE
feat: Send wc_syncDelete request after local storage update

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after: #802

### CODE REVIEW

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.walletconnect.android.internal.utils.MONTH_IN_SECONDS
+import com.walletconnect.foundation.common.model.Topic
+import com.walletconnect.foundation.common.model.Ttl
+import com.walletconnect.sync.common.json_rpc.SyncParams
+import com.walletconnect.sync.common.json_rpc.SyncRpc
+import com.walletconnect.sync.common.model.Store
+import com.walletconnect.sync.storage.StoresStorageRepository
+import com.walletconnect.util.generateId
+
+internal class SendDeleteRequestUseCase(private val jsonRpcInteractor: JsonRpcInteractorInterface, private val storesRepository: StoresStorageRepository) {
```
Comment (TalhaAli00) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: will this use case ever be used anywhere else outside of `DeleteStoreValueUseCase`? If not, why not move this logic into a private function of `DeleteStoreValueUseCase`?

Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: Yeah, we could. If team agrees with you I'm happy to change it ^^

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    // https://github.com/WalletConnect/WalletConnectKotlinV2/issues/800 -> update params to have StoreKey
+    suspend operator fun invoke(key: String, accountId: AccountId, store: Store, onSuccess: () -> Unit, onError: (Throwable) -> Unit) {
+        val storeTopic = runCatching { storesRepository.getStoreTopic(accountId, store) }.fold(onSuccess = { Topic(it) }, onFailure = { error -> return onError(error) })
+        val deleteParams = SyncParams.DeleteParams(key)
+        val payload = SyncRpc.SyncDelete(id = generateId(), params = deleteParams)
```
[the remark this answers is not in the document]
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:22]: I checked and that is how other SDKs declare it.

[the remark this answers is not in the document]
Comment (jakubuid) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:22]: Check again, the PR with the change Talha is talking about was merged yesterday

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt (hunk 1 above)
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: My proposal: https://github.com/WalletConnect/WalletConnectKotlinV2/pull/813

Comment (jakubuid) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: Moving to priv fun makes more sense to me

Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: @jakubuid are you prefering priv fun over inner fun as well?

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:22]: PR with change: https://github.com/WalletConnect/WalletConnectKotlinV2/pull/817

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt (hunk 1 above)
Comment (TalhaAli00) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/requests/outgoing/SendDeleteRequestUseCase.kt:16]: the inner function proposal in #813  works for me

**Review by jakubuid — APPROVED** [review]

**Pull request merged**